beancount/parser/parser_test.py

```diff
@@ -67,6 +69,19 @@ def test_parse_doc__noerrors(self, _, __, ___):
           Assets:US:Cash             -100 USD
         """
 
+    @unittest.skipIf(sys.platform != "win32", reason="test for win32 file separator")
+    def test_win32_filename(self):
+        """filepath in document are expected to be escaped"""
+        row = r'2014-06-08 document Assets:Account1 "D:\a\beancount\beancount\path\to\document.csv"'
+
+        file = io.StringIO()
+
+        print_entries(load_string(row)[0], file=file)
+        assert (
+            load_string(row)[0]
+            != r'2014-06-08 document Assets:Account1 "D:\\a\beancount\beancountpath\todocument.csv"'
```

Comment:
Can you please explain in prose what this test is trying to test?

Unless I am missing something, the test as written now checks that the result of the parse, print, parse round trip gives you a string that is different from the string you started from, which is true in all cases, unless it exists a bug in the parser or in the printer that precisely maps the string you start form to a different string.

Reply:
my bad, shouldnot run printer here

Reply:
is it clear what we are testing now?

Reply:
Now it is clearer. But I don't understand why you think that adding this test is appropriate. The test gives the same result before and after your changes. Do you think it helps extend the test coverage of the parser? Why is this case more important to test than others? Handling of escape sequences is done in the lexer, for which the `filename` field of the `document` directive is just a string like another. I think we agree on the fact that we don't want to change the parser to handle the `filename` field differently than other strings.

Reply:
I cannot know what @blais was asking for. My guess is that he was asking for a test that verifies that the entry printer does not escspe backslashes in the input strings. This test may do that, but it does so in the most counter intuitive way I can imagine and it actually checks that non-escaped backslashes produce unwanted results.

If we want to add a test like this (whch IMHO is not really necessary) I would write it in a very different way: just instantiate a `beancount.core.data.Document` object wit escaped backslashes in the file path, serialize it with `beancount.parser.printer.print_entry()` and check that the serialized file path has exactly the same form. If you craft the file path in the proper way, the test does not need to be platform dependent.

This test would need to live with the tests for the printer, though. Not with the tests for the parser.